skimage/filters/ridges.py

```diff
@@ -0,0 +1,651 @@
+
+            # Store results in (n+1)D matrices
+            filtered_array[i] = filtered
+
+        else:
+
+            # Check image dimensions
+            assert_nD(image, [2, 3])
+
+    # Return for every pixel the value of the (sigma) scale with the maximum
+    # output pixel value
+    return np.max(filtered_array, axis=0)
+
+
+def frangi(image, scale_range=(1, 10), scale_step=2, beta1=None, beta2=None,
+           alpha=0.5, beta=0.5, gamma=15, black_ridges=True):
+    """
+    Filter an image with the Frangi vesselness filter.
+
+    This filter can be used to detect continuous ridges, e.g. vessels,
+    wrinkles, rivers. It can be used to calculate the fraction of the
+    whole image containing such objects.
+
+    Calculates the eigenvectors of the Hessian to compute the similarity of
+    an image region to vessels, according to the method described in _[1].
+
+    Parameters
+    ----------
+    image : (N, M) ndarray
+        Array with input image data.
+    scale_range : 2-tuple of floats, optional
+        The range of sigmas used.
+    scale_step : float, optional
+        Step size between sigmas.
+    alpha : float, optional
+        Frangi correction constant that adjusts the filter's
+        sensitivity to deviation from a plate-like structure.
+    beta = beta1 : float, optional
+        Frangi correction constant that adjusts the filter's
+        sensitivity to deviation from a blob-like structure.
+    gamma = beta2 : float, optional
+        Frangi correction constant that adjusts the filter's
+        sensitivity to areas of high variance/texture/structure.
+    black_ridges : boolean, optional
+        When True (the default), the filter detects black ridges; when
+        False, it detects white ridges.
+
+    Returns
+    -------
+    out : (N, M) ndarray
+        Filtered image (maximum of pixels across all scales).
+
+    Notes
+    -----
+    Written by Marc Schrijver, November 2001
+    Re-Written by D. J. Kroon University of Twente, May 2009, _[2]
+    Adoption of 3D version from D. G. Ellis, Januar 20017, _[3]
+
+    References
+    ----------
+    .. [1] Frangi, A. F., Niessen, W. J., Vincken, K. L., & Viergever, M. A.
+        (1998,). Multiscale vessel enhancement filtering. In International
+        Conference on Medical Image Computing and Computer-Assisted
+        Intervention (pp. 130-137). Springer Berlin Heidelberg.
+        http://www.tecn.upf.es/~afrangi/articles/miccai1998.pdf
+    .. [2] Kroon, D. J.: Hessian based Frangi vesselness filter.
+    .. [3] Ellis, D. G.: https://github.com/ellisdg/frangi3d/tree/master/frangi
+    """
+
+    # Check deprecated keyword parameters
+    if beta1:
+        warn("""Use keyword parameter 'beta' instead of 'beta1' which
+                will be removed in version 0.16.""")
+        beta = beta1
+
+    if beta1:
+        warn("""Use keyword parameter 'gamma' instead of 'beta2' which
+                will be removed in version 0.16.""")
```

Comment:
This should be 0.17 now, *and* this shouldn't be a triple-quoted string (which will preserve the whitespace) but a string continuation:

```python
warn('The start of the message and '
     'the end of the message')
```

Reply:
Check